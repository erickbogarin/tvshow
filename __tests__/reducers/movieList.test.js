import { ActionTypes } from 'constants/index';
import movies from 'reducers/movieList';

test('should return initial state', () => {
  expect(movies(undefined, { type: {} })).toMatchSnapshot();
});

test('should handle SET_MOVIES with payload', () => {
  const currentState = movies(undefined, { type: {}}),
    payload = { results: [{ title: 'a' }] };

  const nextState = movies(undefined, { type: ActionTypes.SET_MOVIES, payload});
  
  expect(currentState).not.toEqual(nextState);
  expect(nextState).toMatchSnapshot();
});

test('should handle RESET_SEARCH', () => {
  const payload = { movies: [{ title: 'a' }] },
    currentState = movies(undefined, { type: ActionTypes.SET_MOVIES, payload}); 
  
  const nextState = movies(undefined, { type: ActionTypes.RESET_SEARCH});
    
  expect(nextState).toMatchSnapshot();
});

