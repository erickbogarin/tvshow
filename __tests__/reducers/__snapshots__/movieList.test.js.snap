exports[`test should handle RESET_SEARCH 1`] = `
Object {
  "movies": Array [],
  "page": 0,
  "total_pages": 0,
  "total_results": 0,
}
`;

exports[`test should handle SET_MOVIES with payload 1`] = `
Object {
  "movies": Array [
    Object {
      "title": "a",
    },
  ],
  "page": 0,
  "total_pages": 0,
  "total_results": 0,
}
`;

exports[`test should return initial state 1`] = `
Object {
  "movies": Array [],
  "page": 0,
  "total_pages": 0,
  "total_results": 0,
}
`;
